plutus-tx/src/PlutusTx/Numeric.hs

```diff
@@ -55,15 +56,15 @@ newtype Additive a = Additive a
 
 instance Semigroup a => AdditiveSemigroup (Additive a) where
     {-# INLINABLE (+) #-}
-    Additive x + Additive y = Additive (x <> y)
+    (+) = coerce ((<>) :: a -> a -> a)
```

Comment:
Is it more efficient this way?

Reply:
It's an example of eta-reducing stuff. I'm also unsure why the eta-reduction helps except insofar as it shrinks the size of terms, which maybe makes the inliner more likely to inline things. For cases like this it means we'll end up with a binding which is just a variable reference, which we'll almost certainly inline, I hope.

Reply:
> Is it more efficient this way?

Yes, fewer lambdas and fewer applications.

It even helps in Haskell, which is why we do it all over the place on the evaluation path. Quoting a Note from `base`:

```
{-
Note [Function coercion]
~~~~~~~~~~~~~~~~~~~~~~~

Several functions here use (#.) instead of (.) to avoid potential efficiency
problems relating to #7542. The problem, in a nutshell:

If N is a newtype constructor, then N x will always have the same
representation as x (something similar applies for a newtype deconstructor).
However, if f is a function,

N . f = \x -> N (f x)

This looks almost the same as f, but the eta expansion lifts it--the lhs could
be _|_, but the rhs never is. This can lead to very inefficient code.  Thus we
steal a technique from Shachaf and Edward Kmett and adapt it to the current
(rather clean) setting. Instead of using  N . f,  we use  N #. f, which is
just

coerce f `asTypeOf` (N . f)

That is, we just *pretend* that f has the right type, and thanks to the safety
of coerce, the type checker guarantees that nothing really goes wrong. We still
have to be a bit careful, though: remember that #. completely ignores the
*value* of its left operand.
-}
```

Also what Michael said.